app/main.py

```diff
@@ -1,6 +1,20 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def inner(*args: tuple, **kwargs) -> Any:
+        key = (func.__name__, *args, tuple(kwargs))
+
+        if key in cache_dict.keys():
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            cache_dict[key] = func(*args, **kwargs)
```

Comment:
```suggestion
        if key in cache_dict:
            print("Getting from cache")
        else:
            print("Calculating new result")
            cache_dict[key] = func(*args, **kwargs)
```
`keys()` used by default